src/runtime/pkg/kata-monitor/cri.go

```diff
@@ -117,13 +117,12 @@ func parseEndpoint(endpoint string) (string, string, error) {
 	}
 }
 
-// getSandboxes gets kata sandboxes from the container engine.
-func (km *KataMonitor) getSandboxes() (map[string]struct{}, error) {
-
-	sandboxMap := make(map[string]struct{})
+// getSandboxes gets ready sandboxes from the container engine and returns an updated sandboxMap
+func (km *KataMonitor) getSandboxes(sandboxMap map[string]bool) (map[string]bool, error) {
+	newMap := make(map[string]bool)
 	runtimeClient, runtimeConn, err := getRuntimeClient(km.runtimeEndpoint)
 	if err != nil {
-		return sandboxMap, err
+		return newMap, err
```

Comment:
Why do you return an empty map on error and not `sandboxMap` (which IIUC contains the earlier cache content)? Couldn't you have a transient error from `getRuntimeClient` that would not be worth invalidating the cache?

(Probably does not matter much since you return an error anwyay)

Reply:
Well, when getSandboxes() returns (in startPodCacheUpdater()), if there is an error, the cache is not updated (the cache is updated only when calling sandboxCache.set())
So, the cache is not invalidated.